pkg/keyspace/tso_keyspace_group.go

```diff
@@ -352,6 +360,32 @@ func (m *GroupManager) saveKeyspaceGroups(keyspaceGroups []*endpoint.KeyspaceGro
 	})
 }
 
+// saveGroupWithUpdateKeyspace will try to save the given keyspace groups into the storage.
+// It only update the keyspace field for the keyspace group.
+func (m *GroupManager) saveGroupWithUpdateKeyspace(keyspaceGroup *endpoint.KeyspaceGroup) error {
+	return m.store.RunInTxn(m.ctx, func(txn kv.Txn) error {
+		oldKG, err := m.store.LoadKeyspaceGroup(txn, keyspaceGroup.ID)
+		if err != nil {
+			return err
+		}
+		if oldKG.IsSplitting() {
+			return ErrKeyspaceGroupInSplit
+		}
+		newKG := oldKG
+		newKG.Keyspaces = keyspaceGroup.Keyspaces
+		if oldKG.IsSplitting() {
```

Comment:
It has already checked that "if oldKG.IsSplitting() is true and exit on condition met" at line 371 above.
So the check here is always true.

Reply:
It looks like we have conflict logic here -- if oldKG.IsSplitting() is true, do we want to return ErrKeyspaceGroupInSplit, or want to set newKG.SplitState and proceed?

Reply:
The whole function doesn't provide CAS semantics, which is my real concern. 
Before the caller calls saveGroupWithUpdateKeyspace, the caller must have loaded the keyspace group info from the etcd, then do some calculation and update keyspace list in the memory, finally call saveGroupWithUpdateKeyspace here. If somehow the latest keyspace list loaded here is different from the original one, it will overwrite something and cause inconsistency. Two ways to solve it:
1. saveGroupWithUpdateKeyspace should check if the latest keyspace group info loaded here has the same keypace list as the original one which can pass to this function.
2. Just pass two lists to saveGroupWithUpdateKeyspace -- one keyspace list to add and one keyspace list to delete, and this function just needs to add if it doesn't exist or delete if it exists based on the latest loaded keyspace list here.

Reply:
saveGroupWithUpdateKeyspace is a part of saveKeyspaceGroups, and only update keyspace list param

Reply:
because updateKeyspaceForGroupLocked load kg from memory rather than etcd, so we should not update other params besides keyspace list.

Reply:
keyspace check is in updateKeyspaceForGroupLocked